### PR TITLE
POC: hostmetrics connector

### PR DESCRIPTION
**Pull request by jcreixell**
- Host metrics connector reads resource attributes from traces (and potentially other telemetry signals) and extracts the host information based on a list of predefined attributes, listed in order of priority.
  - It includes an otel collector connector and a wrapping flow component.

### CODE REVIEW

File: component/otelcol/connector/spanhostmetrics/spanhostmetricsconnector.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+						dpCalls.SetIntValue(int64(1))
+
+						// TODO: store the metric in a map and flush it after a certain time instead of on every span
+						err := c.metricsConsumer.ConsumeMetrics(ctx, metrics)
+						if err != nil {
+							return err
+						}
+						break
+					}
+				}
```
Comment (jcreixell) [inline component/otelcol/connector/spanhostmetrics/spanhostmetricsconnector.go:121]: we probably need some telemetry to detect traces with no host info (or at least some logging)

**Comment by rlankfo** [comment]
Awesome, thanks for providing this. One thing to note is this connector could be expanded in the future to include other signals like logs and traces. This makes me think we shouldn't tie the name to a certain signal.

**Comment by jcreixell** [comment]
@rlankfo good point, I thought we could have a separate metric per signal, but the name of the connector should be generic :+1:

**Comment by jcreixell** [comment]
Superseded by #6410

**Pull request closed without merging**